Plutarch/TryFrom.hs

```diff
@@ -318,25 +329,6 @@ instance {-# OVERLAPPING #-} SumValidation n '[] where
 
 ----------------------- other utility functions -----------------------------------------
 
-{- | if there is an instance to recover something that is unwrapped from something that is
- unwrapped, then there is also the possibility to recover the whole thing but wrapped
--}
-instance
-  ( PTryFrom a b
-  , PIsData a
-  , PIsData b
-  ) =>
-  PTryFrom (PAsData a) (PAsData b)
```

Comment:
This was illegal. Even if it's true that `PSubtype a b`, this does **not** imply `PSubtype (PAsData a) (PAsData b)`, and was also a bug before this change!

Reply:
Notably, you could have `PInteger` and `PNatural` have completely different `PIsData` implementations.

Reply:
We don't need to remove it completely, we should just not punsafeCoerce the opq but pdata the uwrapped version. it's probably helpful in many cases anyway because semantically it's correct, isn't it?

Reply:
That would be just as illegal as it violates the laws of `PTryFrom`.

Reply:
That's a pity, it shouldn't really break the laws of PTryFrom tbh, I think many people would semantically expect that 
PSubType (PAsData PInteger) (PAsData PNatural). But I guess they can just add this instance themselves.

Reply:
You can have an instance in specifically that case, but it's not really possible to have a general purpose reasonable law that allows such a general instance. The return of `ptryFrom` *must* be the same as the input, with only the type changed.

Reply:
And anyway, it would still violate the `PSubtype` constraint.

Reply:
Wait, so is there no "easy" way for users to get a `PTryFrom` instance for types like `CurrencySymbol` (and other "variant-less" newtypes)?

Reply:
See comment by Emily at #438

Reply:
Yeah that looks good - we should have that instance if we don't already.